src/gui/qgstemporalcontrollerwidget.cpp

```diff
@@ -574,7 +575,12 @@ void QgsTemporalControllerWidget::setDates( const QgsDateTimeRange &range )
     whileBlocking( mEndDateTime )->setDateTime( range.end() );
     whileBlocking( mFixedRangeStartDateTime )->setDateTime( range.begin() );
     whileBlocking( mFixedRangeEndDateTime )->setDateTime( range.end() );
-    updateTemporalExtent();
+    // only if range is different from mNavigationObject->temporalExtents update
+    // else recursion as updateTemporalExtent itself calls mNavigationObject->setTemporalExtents
+    if ( !( range == mNavigationObject->temporalExtents() ) )
```

Comment:
```suggestion
    if ( range != mNavigationObject->temporalExtents() )
```

I think that's easier to read